generated/nidcpower/nidcpower/lcr_load_compensation_spot.py

```diff
@@ -0,0 +1,118 @@
+    if platform.system() == "Windows" and platform.architecture()[0] == "64bit":
+        _pack_ = 8
+    elif platform.system() == "Linux" or platform.system() == "Darwin":
+        _pack_ = 4
+    else:
+        _pack_ = 1
+    _fields_ = [
+        ("frequency", nidcpower._visatype.ViReal64),
+        ("reference_value_type", nidcpower._visatype.ViInt32),
+        ("reference_value_a", nidcpower._visatype.ViReal64),
+        ("reference_value_b", nidcpower._visatype.ViReal64),
+    ]
+
+    def __init__(self, data):
+        super(ctypes.Structure, self).__init__()
+        self.frequency = data.frequency
+        self.reference_value_type = enums.LCRReferenceValueType(data.reference_value_type).value
+        if self.reference_value_type == enums.LCRReferenceValueType.IMPEDANCE.value:
+            self.reference_value_a = data.reference_value.real
+            self.reference_value_b = data.reference_value.imag
+        else:
+            self.reference_value_a = data.reference_value
+            self.reference_value_b = 0.0
+
+
+class LCRLoadCompensationSpot(object):
+    """Constructs a DUT specification for a given frequency to use in LCR load compensation."""
+
+    def __init__(
+        self,
+        frequency=0.0,
+        impedance=None,
+        ideal_capacitance=None,
+        ideal_inductance=None,
+        ideal_resistance=None,
+    ):
+        """LCRLoadCompensationSpot
+
+        Creates and returns an LCRLoadCompensationSpot object. At most one of impedance,
+            ideal_capacitance, ideal_inductance and ideal_resistance can be set, and the remaining
+            parameters (excluding frequency) must be None. The parameter that is not None specifies
+            the known specification value of the DUT to be used as the basis for load compensation.
+            If all of them are None, then the default value of `impedance=complex()` will be used.
+
+        Args:
+            frequency (float): Specifies the spot frequency, in Hz.
+
+            impedance (complex): Specifies the actual impedance of your DUT to be used as the basis
+                for load compensation, or None to use another type of DUT specification value.
+
+            ideal_capacitance (float): Specifies the ideal capacitance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+
+            ideal_inductance (float): Specifies the ideal inductance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+
+            ideal_resistance (float): Specifies the ideal resistance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+        """
+        self.frequency = frequency
+        # Set default values
+        self.reference_value_type = enums.LCRReferenceValueType.IMPEDANCE
+        self.reference_value = complex()
```

Comment:
Giving default values to custom type members is actually based on what the other existing custom types in other drivers do (e.g. https://github.com/ni/nimi-python/blob/master/src/niscope/custom_types/measurement_stats.py#L2).

Nevertheless, since we are going with the approach of having no unnecessary default values here, I have removed the setting of default values in my latest commit. I have also updated the input validation code to throw an error if all the 4 input parameters (excluding `frequency`) are None.